openwisp_monitoring/device/api/views.py

```diff
@@ -203,6 +205,72 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        if 'resources' not in data:
+            return
+        self._write_cpu(data['resources'], pk, ct)
+        self._write_disk(data['resources'], pk, ct)
+        self._write_memory(data['resources'], pk, ct)
+
+    def _write_cpu(self, resources, primary_key, content_type):
+        extra_values = {
+            'load_1': float(resources['load'][0]),
```

Comment:
same here, we're assuming `load` is present, but `load` is not flagged as required in the schema.
So if the monitoring data doesn't include `load`, this will generate an exception and no metric will be collected.

We could make `load` mandatory, but I think not making it mandatory gives us flexibility.